azurerm/internal/services/cosmos/cosmosdb_cassandra_table_resource.go

```diff
@@ -0,0 +1,421 @@
+				ValidateFunc: validate.CosmosAccountName,
+			},
+
+			"default_ttl": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntAtLeast(-1),
+			},
+
+			"keyspace_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.CosmosEntityName,
+			},
+
+			"schema": common.CassandraTableSchemaPropertySchema(),
+
+			"throughput": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validate.CosmosThroughput,
+			},
+
+			"autoscale_settings": common.DatabaseAutoscaleSettingsSchema(),
+		},
+	}
+}
+
+func resourceCosmosDbCassandraTableCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Cosmos.CassandraClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	account := d.Get("account_name").(string)
+	keyspace := d.Get("keyspace_name").(string)
+
+	existing, err := client.GetCassandraTable(ctx, resourceGroup, account, keyspace, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("Error checking for presence of creating Cosmos Cassandra Table %q (Account: %q, Keyspace: %q): %+v", name, account, keyspace, err)
+		}
+	} else {
+		if existing.ID == nil && *existing.ID == "" {
+			return fmt.Errorf("Error generating import ID for Cosmos Cassandra Table %q (Account: %q, Keyspace: %q)", name, account, keyspace)
+		}
+
+		return tf.ImportAsExistsError("azurerm_cosmosdb_cassandra_table", *existing.ID)
+	}
+
+	table := documentdb.CassandraTableCreateUpdateParameters{
+		CassandraTableCreateUpdateProperties: &documentdb.CassandraTableCreateUpdateProperties{
+			Resource: &documentdb.CassandraTableResource{
+				ID: &name,
+			},
+			Options: &documentdb.CreateUpdateOptions{},
+		},
+	}
+
+	table.CassandraTableCreateUpdateProperties.Resource.Schema = expandTableSchema(d)
+
+	if defaultTTL, hasTTL := d.GetOk("default_ttl"); hasTTL {
+		table.CassandraTableCreateUpdateProperties.Resource.DefaultTTL = utils.Int32(int32(defaultTTL.(int)))
+	}
+
+	if throughput, hasThroughput := d.GetOk("throughput"); hasThroughput {
+		if throughput != 0 {
+			table.CassandraTableCreateUpdateProperties.Options.Throughput = common.ConvertThroughputFromResourceData(throughput)
+		}
+	}
+
+	if _, hasAutoscaleSettings := d.GetOk("autoscale_settings"); hasAutoscaleSettings {
+		table.CassandraTableCreateUpdateProperties.Options.AutoscaleSettings = common.ExpandCosmosDbAutoscaleSettings(d)
+	}
+
+	future, err := client.CreateUpdateCassandraTable(ctx, resourceGroup, account, keyspace, name, table)
+	if err != nil {
+		return fmt.Errorf("Error issuing create/update request for Cosmos Cassandra Table %q (Account: %q, Keyspace: %q): %+v", name, account, keyspace, err)
```

Comment:
fwiw the description can come from the ID here:

```suggestion
			return fmt.Errorf("creating %s: %+v", *id, err)
```